### PR TITLE
Config file: add support for latest aliases

### PR DESCRIPTION
**Pull request by stsewd**
- Closes https://github.com/readthedocs/readthedocs.org/issues/8861
- Closes https://github.com/readthedocs/readthedocs.org/issues/10912
- Closes https://github.com/readthedocs/readthedocs.org/issues/10401

Ref https://github.com/readthedocs/meta/discussions/140

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11081.org.readthedocs.build/en/11081/

- Developer's documentation (`dev`): https://dev--11081.org.readthedocs.build/en/11081/

### CODE REVIEW

File: docs/user/config-file/v2.rst [hunk 1]
```diff
@@ -286,6 +295,15 @@ Version specifiers for each tool. It must contain at least one tool.
 :Options: ``python``, ``nodejs``, ``rust``, ``golang``
 :Required: ``true``
 
+.. note::
+
+   Each tool has a ``latest`` option available, which refers to the latest version available on Read the Docs,
+   not the latest version available. Versions are updated at least once every six months.
```
Comment (humitos) [inline docs/user/config-file/v2.rst:301]:
```suggestion
   Each tool has a ``latest`` option available, which refers to the latest version available on Read the Docs,
   which may not match with the latest version officially released by the tool.
   `latest` versions are updated every 6 months to point to the latest version officially released.
```

File: docs/user/config-file/v2.rst [hunk 2]
```diff
@@ -286,6 +295,15 @@ Version specifiers for each tool. It must contain at least one tool.
...
+
+.. warning::
+
+   Using ``latest`` may break your builds unexpectedly if your project isn't compatible with the latest version of the tool when it's updated.
```
Comment (humitos) [inline docs/user/config-file/v2.rst:305]:
```suggestion
   Using ``latest`` may break your builds unexpectedly if your project isn't compatible with the newest version version of the tool released when Read the Docs upgrades to it.
```

File: readthedocs/builds/constants_docker.py [hunk 3]
```diff
@@ -9,17 +9,16 @@
...
-# development and production environments to compile and cache the new
-# tool/version
-#
-# Note that when updating this options, you should also update the file:
-# readthedocs/rtd_tests/fixtures/spec/v2/schema.json
+# - Add a mapping between the expected version in the config file, to the full
+#   version installed via asdf (found via ``asdf list all <tool>``).
+# - Run the script ``./scripts/compile_version_upload.sh`` in
+#   development and production environments to compile and cache the new
+#   tool/version.
```
Comment (humitos) [inline readthedocs/builds/constants_docker.py:18]:
```suggestion
# - Run the script ``./scripts/compile_version_upload.sh`` in
#   development to compile and cache the new tool/version.
```

File: readthedocs/builds/constants_docker.py [hunk 4]
```diff
@@ -9,17 +9,16 @@
...
+# - Update the latest aliases for OS and tools (below this setting).
```
Comment (humitos) [inline readthedocs/builds/constants_docker.py:19]:
```suggestion
# - Update the CircleCI job on ``readthedocs-docker-images`` repository with the newer versions (https://github.com/rtfd/readthedocs-docker-images/blob/d2760526abdfe27001946614b749abf8011b7f90/.circleci/config.yml#L38-L44)
# - Update the latest aliases for OS and tools (below this setting).
```

File: docs/user/config-file/v2.rst [hunk 5]
```diff
@@ -274,9 +274,18 @@ Image names refer to the operating system Read the Docs uses to build them.
    Arbitrary Docker images are not supported.
 
 :Type: ``string``
-:Options: ``ubuntu-20.04``, ``ubuntu-22.04``
+:Options: ``ubuntu-20.04``, ``ubuntu-22.04``, ``ubuntu-latest-lts``
```
[the remark this answers is not in the document]
Comment (stsewd) [reply docs/user/config-file/v2.rst:277]: @ericholscher any preference?

[the remark this answers is not in the document]
Comment (ericholscher) [reply docs/user/config-file/v2.rst:277]: Will we have an `ubuntu-latest` and isn't an LTS?

File: docs/user/config-file/v2.rst [hunk 6]
```diff
@@ -286,6 +295,16 @@ Version specifiers for each tool. It must contain at least one tool.
 :Options: ``python``, ``nodejs``, ``rust``, ``golang``
 :Required: ``true``
 
+.. note::
+
+   Each tool has a ``latest`` option available, which refers to the latest version available on Read the Docs,
+   which may not match the latest version officially released.
+   Versions and the ``latest`` option are updated at least once every six months to keep up with the latest releases.
```
Comment (ericholscher) [inline docs/user/config-file/v2.rst:302]: Are we documenting what `latest` currently points to? Seems like that would be useful information, but also a pain to update.

File: docs/user/config-file/v2.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/config-file/v2.rst:277]: > Will we have an `ubuntu-latest` that isn't an LTS?

No, we don't plan to do that. (see https://github.com/readthedocs/readthedocs.org/issues/8861#issuecomment-1026168933)

> But I agree with Manuel that `ubuntu-lts-latest` matches more what we're using other places.

👍🏼 --  We talked about this at https://github.com/readthedocs/readthedocs.org/issues/8861#issuecomment-1302287669

File: docs/user/config-file/v2.rst (hunk 6 above)
Comment (stsewd) [reply docs/user/config-file/v2.rst:302]: Was hoping that `(alias for the latest version available on Read the Docs)` on each tag would be enough given the list of all available versions.

Comment (ericholscher) [reply docs/user/config-file/v2.rst:302]: So it's basically the same as the first item in the list?

Comment (stsewd) [reply docs/user/config-file/v2.rst:302]: Last item

**Comment by stsewd** [comment]
Feel free to merge if you think this should go out with tomorrows deploy!

**Pull request merged**